types.go

```diff
@@ -298,6 +298,24 @@ type SignedTreeHead struct {
 	LogID             SHA256Hash      `json:"log_id"`              // The SHA256 hash of the log's public key
 }
 
+func (s SignedTreeHead) String() string {
+	sigStr, err := s.TreeHeadSignature.Base64String()
+	if err != nil {
+		sigStr = tls.DigitallySigned(s.TreeHeadSignature).String()
```

Comment:
What's the reason to have 2 formats? Maybe just use `String()` unconditionally if it's not very ugly?

Reply:
I wanted to have the Base64String() version of the signature, as it is then the signature in the form that is in the JSON that is returned from the Log, so the String() version of the SignedTreeHead will contain all of the fields as they would have been received from the Log.

However, that Base64String() function can return an error, and a String() function shouldn't, so the DigitallySigned.String is a fallback - it prints it in hex instead, with a bunch of additional info, so (in my opinion at least) is quite a bit uglier.